crowbar_framework/app/models/dns_service.rb

```diff
@@ -91,38 +96,43 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     return if all_nodes.empty?
 
     tnodes = role.override_attributes["dns"]["elements"]["dns-server"]
-    nodes = tnodes.map {|n| NodeObject.find_node_by_name n}
-
-    # electing master dns-server
-    master = nil
-    admin = nil
-    nodes.each do |node|
-      if node[:dns][:master]
-        master = node
-        break
-      elsif node.admin?
-        admin = node
+
+    if !tnodes.blank?
+      nodes = tnodes.map {|n| NodeObject.find_node_by_name n}
+      # electing master dns-server
+      master = nil
+      admin = nil
+      nodes.each do |node|
+        if node[:dns][:master]
+          master = node
+          break
+        elsif node.admin?
+          admin = node
+        end
       end
-    end
-    if master.nil?
-      unless admin.nil?
-        master = admin
-      else
-        master = nodes.first
+      if master.nil?
+        unless admin.nil?
+          master = admin
+        else
+          master = nodes.first
+        end
       end
-    end
 
-    slave_ips = nodes.map {|n| n[:crowbar][:network][:admin][:address]}
-    slave_ips.delete(master[:crowbar][:network][:admin][:address])
-    slave_nodes = tnodes.dup
-    slave_nodes.delete(master.name)
-
-    nodes.each do |node|
-      node.set[:dns][:master_ip] = master[:crowbar][:network][:admin][:address]
-      node.set[:dns][:slave_ips] = slave_ips
-      node.set[:dns][:slave_names] = slave_nodes
-      node.set[:dns][:master] = (master.name == node.name)
-      node.save
+      slave_ips = nodes.map {|n| n[:crowbar][:network][:admin][:address]}
+      slave_ips.delete(master[:crowbar][:network][:admin][:address])
+      slave_nodes = tnodes.dup
+      slave_nodes.delete(master.name)
+
+      nodes.each do |node|
+        node.set[:dns][:master_ip] = master[:crowbar][:network][:admin][:address]
+        node.set[:dns][:slave_ips] = slave_ips
+        node.set[:dns][:slave_names] = slave_nodes
+        node.set[:dns][:master] = (master.name == node.name)
+        node.save
+      end
+    else
+      nameservers = role.default_attributes["dns"]["nameservers"]
+      return if nameservers.blank?
```

Comment:
What is this ```else``` branch good for? I might be overlooking something but aren't  we at the end of the function here anyway, why return? And if we don't need the return we don't need the ```else``` as well.